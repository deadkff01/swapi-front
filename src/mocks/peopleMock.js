export const peopleMock = [
  {
    name: 'Luke Skywalker',
    height: '172',
    mass: '77',
    hair_color: 'blond',
    skin_color: 'fair',
    eye_color: 'blue',
    birth_year: '19BBY',
    gender: 'male',
    homeworld: 'http://swapi.dev/api/planets/1/',
    films: [
      'http://swapi.dev/api/films/1/',
      'http://swapi.dev/api/films/2/',
      'http://swapi.dev/api/films/3/',
      'http://swapi.dev/api/films/6/'
    ],
    species: [],
    vehicles: [
      'http://swapi.dev/api/vehicles/14/',
      'http://swapi.dev/api/vehicles/30/'
    ],
    starships: [
      'http://swapi.dev/api/starships/12/',
      'http://swapi.dev/api/starships/22/'
    ],
    created: '2014-12-09T13:50:51.644000Z',
    edited: '2014-12-20T21:17:56.891000Z',
    url: 'http://swapi.dev/api/people/1/'
  },
  {
    name: 'C-3PO',
    height: '167',
    mass: '75',
    hair_color: 'n/a',
    skin_color: 'gold',
    eye_color: 'yellow',
    birth_year: '112BBY',
    gender: 'n/a',
    homeworld: 'http://swapi.dev/api/planets/1/',
    films: [
      'http://swapi.dev/api/films/1/',
      'http://swapi.dev/api/films/2/',
      'http://swapi.dev/api/films/3/',
      'http://swapi.dev/api/films/4/',
      'http://swapi.dev/api/films/5/',
      'http://swapi.dev/api/films/6/'
    ],
    species: ['http://swapi.dev/api/species/2/'],
    vehicles: [],
    starships: [],
    created: '2014-12-10T15:10:51.357000Z',
    edited: '2014-12-20T21:17:50.309000Z',
    url: 'http://swapi.dev/api/people/2/'
  },
  {
    name: 'R2-D2',
    height: '96',
    mass: '32',
    hair_color: 'n/a',
    skin_color: 'white, blue',
    eye_color: 'red',
    birth_year: '33BBY',
    gender: 'n/a',
    homeworld: 'http://swapi.dev/api/planets/8/',
    films: [
      'http://swapi.dev/api/films/1/',
      'http://swapi.dev/api/films/2/',
      'http://swapi.dev/api/films/3/',
      'http://swapi.dev/api/films/4/',
      'http://swapi.dev/api/films/5/',
      'http://swapi.dev/api/films/6/'
    ],
    species: ['http://swapi.dev/api/species/2/'],
    vehicles: [],
    starships: [],
    created: '2014-12-10T15:11:50.376000Z',
    edited: '2014-12-20T21:17:50.311000Z',
    url: 'http://swapi.dev/api/people/3/'
  }
];

export const C3PO = {
  name: 'C-3PO',
  height: '167',
  mass: '75',
  hair_color: 'n/a',
  skin_color: 'gold',
  eye_color: 'yellow',
  birth_year: '112BBY',
  gender: 'n/a',
  homeworld: 'http://swapi.dev/api/planets/1/',
  films: [
    'http://swapi.dev/api/films/1/',
    'http://swapi.dev/api/films/2/',
    'http://swapi.dev/api/films/3/',
    'http://swapi.dev/api/films/4/',
    'http://swapi.dev/api/films/5/',
    'http://swapi.dev/api/films/6/'
  ],
  species: ['http://swapi.dev/api/species/2/'],
  vehicles: [],
  starships: [],
  created: '2014-12-10T15:10:51.357000Z',
  edited: '2014-12-20T21:17:50.309000Z',
  url: 'http://swapi.dev/api/people/2/'
};
