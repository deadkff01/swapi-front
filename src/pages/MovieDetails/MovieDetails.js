import { useEffect, useState, useCallback } from 'react';
import { Container } from 'components/Container';
import { Title } from 'components/Title';
import { PosterImage } from 'components/PosterImage';
import { useHttpInstance } from 'services/baseService';
import { useParams } from '@reach/router';
import { CharacterList } from 'components/CharacterList';
import { Loader } from 'components/Loader';
import { ErrorWarning } from 'components/ErrorWarning';
import { Fade } from 'utils/Fade';
import { BackButton } from 'components/BackButton';

export default function MovieDetails() {
  const [movie, setMovie] = useState({});
  const params = useParams();
  const { get, response, error, loading } = useHttpInstance();

  const getMovie = useCallback(async () => {
    const movie = await get(`movies/${params.id}`);
    if (response.ok) setMovie(movie);
  }, [get, response, params.id]);

  useEffect(() => {
    getMovie();
  }, [getMovie]);

  if (loading) {
    return <Loader />;
  }

  if (error) {
    return (
      <ErrorWarning message="Error, verify your conection or reload the page! may the force be with you!" />
    );
  }

  return (
    <Fade in={true} unmountOnExit appear>
      <Container
        data-testid="container"
        className="flex flex-wrap justify-center"
      >
        <BackButton />
        <div className="w-full text-center px-5 mb-5">
          <Title className="text-4xl" align="center">
            {movie.title}
          </Title>
          <PosterImage className={`movie-image-${movie.episode_id} mx-auto`} />
        </div>
        <div className="lg:flex ">
          <div className="md:w-full lg:w-2/4 text-left text-lg px-5 mb-10">
            <Title className="text-3xl" align="left">
              Opening Crawl
            </Title>
            <div>{movie.opening_crawl}</div>
          </div>
          <div className="md:w-full lg:w-2/4 text-left px-5  text-lg  mb-10">
            <Title className="text-3xl" align="left">
              Informations
            </Title>
            <div>Director: {movie.director}</div>
            <div>Producer: {movie.producer}</div>
            <div>Release date: {movie.release_date}</div>
          </div>
        </div>

        {movie?.characters?.length > 0 ? (
          <CharacterList characterList={movie.characters} />
        ) : null}
      </Container>
    </Fade>
  );
}
