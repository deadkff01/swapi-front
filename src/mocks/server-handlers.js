import { rest } from 'msw';
import { mockMovies, aNewHopeMock } from './mockMovies';
import { peopleMock, C3PO } from './peopleMock';

const handlers = [
  rest.get(`${process.env.REACT_APP_ENDPOINT}/movies`, (_, res, ctx) =>
    res(ctx.json(mockMovies))
  ),
  rest.get(`${process.env.REACT_APP_ENDPOINT}/movies/1`, (_, res, ctx) =>
    res(ctx.json(aNewHopeMock))
  ),
  rest.post(`${process.env.REACT_APP_ENDPOINT}/people`, (_, res, ctx) =>
    res(ctx.json(peopleMock))
  ),
  rest.get(`${process.env.REACT_APP_ENDPOINT}/people/2`, (_, res, ctx) =>
    res(ctx.json(C3PO))
  )
];

export { handlers };
