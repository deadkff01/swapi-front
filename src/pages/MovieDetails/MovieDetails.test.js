import { cleanup, screen } from '@testing-library/react';
import '../../setupTests';
import { renderWithRouterParams } from '../../testUtils';
import MovieDetails from './MovieDetails';
import '@testing-library/jest-dom/extend-expect';
import { server } from 'mocks/server';

describe('MovieDetails component', () => {
  beforeAll(() => server.listen());

  afterEach(() => server.resetHandlers());

  afterAll(() => {
    jest.useRealTimers();
    server.close();
    cleanup();
  });

  test('renders the component', async () => {
    renderWithRouterParams(<MovieDetails path="/movie/:id" />, {
      route: '/movie/1'
    });
    const container = await screen.findByTestId('container');
    expect(container).toMatchSnapshot();
  });

  test('renders the loader', async () => {
    renderWithRouterParams(<MovieDetails path="/movie/:id" />, {
      route: '/movie/1'
    });
    const loader = await screen.findByTestId('loader');
    expect(loader).toBeDefined();
  });

  test('renders with valid values', async () => {
    renderWithRouterParams(<MovieDetails path="/movie/:id" />, {
      route: '/movie/1'
    });

    const title = await screen.findByText('A New Hope');
    const director = await screen.findByText('Director: George Lucas');
    const producer = await screen.findByText(
      'Producer: Gary Kurtz, Rick McCallum'
    );
    const releaseDate = await screen.findByText('Release date: 1977-05-25');

    expect(title).toBeDefined();
    expect(director).toBeDefined();
    expect(producer).toBeDefined();
    expect(releaseDate).toBeDefined();
  });
});
