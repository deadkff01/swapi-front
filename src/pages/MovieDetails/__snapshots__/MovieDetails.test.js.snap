// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails component renders the component 1`] = `
<div
  class="sc-bdfBwQ bqIkDZ flex flex-wrap justify-center"
  data-testid="container"
>
  <button
    class="sc-iBPRYJ fNSILO mt-5"
    data-testid="movie-card"
  >
    &lt; 
     Back
  </button>
  <div
    class="w-full text-center px-5 mb-5"
  >
    <h1
      class="sc-gsTCUz ipzQyr text-4xl"
    />
    <div
      class="sc-dlfnbm gcQIRF movie-image-undefined mx-auto"
    />
  </div>
  <div
    class="lg:flex "
  >
    <div
      class="md:w-full lg:w-2/4 text-left text-lg px-5 mb-10"
    >
      <h1
        class="sc-gsTCUz hZPDCb text-3xl"
      >
        Opening Crawl
      </h1>
      <div />
    </div>
    <div
      class="md:w-full lg:w-2/4 text-left px-5  text-lg  mb-10"
    >
      <h1
        class="sc-gsTCUz hZPDCb text-3xl"
      >
        Informations
      </h1>
      <div>
        Director: 
      </div>
      <div>
        Producer: 
      </div>
      <div>
        Release date: 
      </div>
    </div>
  </div>
</div>
`;
